Tests/ContentfulBaseTestCase.m

```diff
@@ -37,45 +40,80 @@ @interface ContentfulBaseTestCase ()
-        return nil;
-    }
-    
-    NSMutableArray *testInvocations = [NSMutableArray arrayWithArray:[super testInvocations]];
-    
-    if ([self instancesRespondToSelector:@selector(beforeAll)]) {
-        NSInvocation *beforeAll = [NSInvocation invocationWithMethodSignature:SIG(self, @selector(beforeAll))];
-        beforeAll.selector = @selector(beforeAll);
-        [testInvocations insertObject:beforeAll atIndex:0];
-    }
-    
-    if ([self instancesRespondToSelector:@selector(afterAll)]) {
-        NSInvocation *afterAll = [NSInvocation invocationWithMethodSignature:SIG(self, @selector(afterAll))];
-        afterAll.selector = @selector(afterAll);
-        [testInvocations addObject:afterAll];
-    }
-    
-    return testInvocations;
-}
+- (void)setUp {
+    [super setUp];
 
-#pragma mark -
+    self.cassetteBaseName = NSStringFromClass([self class]);
+    self.client = [CDAClient new];
 
-- (void)afterAll
-{
-    [VCR save:[NSString stringWithFormat:@"/tmp/%@.json", self.cassetteBaseName]];
-    [VCR stop];
+    self.requestReplayManager = [CCLRequestReplayManager new];
+
+    [self.requestReplayManager replay];
+
+    self.snapshotTestController = [[FBSnapshotTestController alloc] initWithTestClass:[self class]];
+    self.snapshotTestController.referenceImagesDirectory = [[NSBundle bundleForClass:[self class]]
+                                                            bundlePath];
+}
+
+-(void)tearDown {
+    [super tearDown];
 
-#ifndef __IPHONE_9_0
-    __gcov_flush();
-#endif
+    [self.requestReplayManager stopReplay];
+    self.requestReplayManager = nil;
 }
 
+//+ (NSArray *)testInvocations
+//{
+//    if (self == [ContentfulBaseTestCase class]) {
+//        return nil;
+//    }
+//    
+//    NSMutableArray *testInvocations = [NSMutableArray arrayWithArray:[super testInvocations]];
+//    
+//    if ([self instancesRespondToSelector:@selector(beforeAll)]) {
+//        NSInvocation *beforeAll = [NSInvocation invocationWithMethodSignature:SIG(self, @selector(beforeAll))];
+//        beforeAll.selector = @selector(beforeAll);
+//        [testInvocations insertObject:beforeAll atIndex:0];
+//    }
+//    
+//    if ([self instancesRespondToSelector:@selector(afterAll)]) {
+//        NSInvocation *afterAll = [NSInvocation invocationWithMethodSignature:SIG(self, @selector(afterAll))];
+//        afterAll.selector = @selector(afterAll);
+//        [testInvocations addObject:afterAll];
+//    }
+//    
+//    return testInvocations;
+//}
+//
+//#pragma mark -
+//
+//- (void)beforeAll
+//{
+//    [VCR loadCassetteWithContentsOfURL:[[NSBundle bundleForClass:[self class]]
+//                                        URLForResource:self.cassetteBaseName withExtension:@"json"]];
+//    [VCR start];
+//}
+//
+//- (void)afterAll
+//{
+//    [VCR save:[NSString stringWithFormat:@"/tmp/%@.json", self.cassetteBaseName]];
+//    [VCR stop];
+//
+//#ifndef __IPHONE_9_0
+//    __gcov_flush();
+//#endif
+//}
```

Comment:
I would just remove the code, if we need to comment it. For me it looks like bad style in an OSS library.